test_pull_requests

```diff
@@ -152,6 +177,89 @@ def submitted_tests_for_branch(branch)
   $submitted_tests[branch] ? $submitted_tests[branch] : $submitted_tests['*']
 end
 
+# Pulls out as much metadata from the bot comment as it can, including
+# state (AKA content), Jenkins build URL, base repo commit ID, and the
+# extended tests metadata
+def extract_bot_comment_metadata(bot_comment_body, settings=nil)
+  test_prefix = nil
+  content = nil
+  build_url = nil
+  base_commit = nil
+  extended_tests = nil
+  state = nil
+  if settings.nil?
+    # Do our best to find the prefix since we don't know it beforehand
+    if bot_comment_body =~ /^([^:(]+): /
+      test_prefix = $1
+    elsif bot_comment_body =~ /^([^ (]+) /
+      test_prefix = $1
+    end
+  else
+    test_prefix = settings['test_prefix']
+  end
+  PR_STATES.each do |pattern, state_name|
+    # Do our best to find the content (aka state) even if there aren't metadata in parens after it
+    if test_prefix && bot_comment_body =~ /^#{test_prefix} ([^(]+) \(/
+      content = $1 || ''
+    elsif test_prefix && bot_comment_body =~ /^#{test_prefix} ([^: ]+):? /
+      content = $1 || ''
+    elsif bot_comment_body =~ pattern
+      state = state_name
+    end
+    if content && !state
+      state = state_name if content =~ pattern
+    end
+    build_url = bot_comment_body[/ \((https?:[^)]+)\)/, 1]
```

Comment:
Yes, I do. I'll fix that.

On Fri, Aug 12, 2016 at 8:47 AM, Dan McPherson <notifications@github.com>
wrote:

> In test_pull_requests
> <https://github.com/openshift/test-pull-requests/pull/31#discussion_r74584373>
> :
>
> > +  else
> > +    test_prefix = settings['test_prefix']
> > +  end
> > +  PR_STATES.each do |pattern, state_name|
> > +    # Do our best to find the content (aka state) even if there aren't metadata in parens after it
> > +    if test_prefix && bot_comment_body =~ /^#{test_prefix} ([^(]+) \(/
> > +      content = $1 || ''
> > +    elsif test_prefix && bot_comment_body =~ /^#{test_prefix} ([^: ]+):? /
> > +      content = $1 || ''
> > +    elsif bot_comment_body =~ pattern
> > +      state = state_name
> > +    end
> > +    if content && !state
> > +      state = state_name if content =~ pattern
> > +    end
> > +    build_url = bot_comment_body[/ \((https?:[^)]+)\)/, 1]
>
> You know the jenkins url right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/test-pull-requests/pull/31/files/f6be253ba3714952ed04378e0054454993bcca3b#r74584373>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABd0FCkBYctwulpidjoSWOUXaHvAtihKks5qfGtEgaJpZM4JcBHJ>
> .
>